docs/source/contributing/development.md

```diff
@@ -0,0 +1,451 @@
+built through previously added boilerplate. (e.g. some random
+[robots](https://github.com/cmin764/robots))
+
+Or maybe you want to test the in-development behaviour through an existing Portal
+[example](https://robocorp.com/portal/robot/robocorp/example-document-ai).
+
+> The test robot can be either in Python or Robot Framework, it doesn't matter, as it
+will use the very same logic from the library.
+
+Therefore, you have two options:
+1. Use a pre-release wheel built and pushed in our own test
+   [DevPI](https://devpi.robocorp.cloud/ci/test).
+   - Use `inv build.publish --ci` to build and publish the package into such DevPI URL.
+   - Go to the URL above and copy the wheel direct link and prefix it with
+     our public read-only `<usr>:<pwd>@` credentials so the wheel is accessible for
+     download.
+   - Paste such link into the conda.yaml of the test robot, therefore the next time you
+     rebuild the holotree env, it will use the code you just pushed, and you'll be able
+     to test the new behaviour in a production-like setup.
+   - Such final test is important before hitting the "Merge" button, as you'll know for
+     sure if the library will work as expected. (and that's the purpose of testing, to
+     do it before merging the code)
+2. Run the test robot with the package's virtual environment.
+   - `cd` into the package you develop the changes into.
+   - `poetry shell` to get a shell within the afferent Poetry _.venv_.
+   - Now `cd` into the test robot directory and run it with:
+     - `python -m robot -L TRACE -d ./output --task "<Task Name>" tasks.robot` if is a
+       RF robot.
+     - `python task.py` for a Python robot.
+   - Getting such a shell will save you time from running commands with
+     `poetry run python ...` while getting the absolute paths rights.
+   - Every new change brought in the project belonging to the active venv will be
+     reflected in the next run of the testing robot, so you won't waste time with
+     building and pushing wheels at every step.
+
+> In order to include more cross-dependent in-development packages under the same venv,
+you have to use the `install.install-local` & `install.reset-local` helpers. (e.g.
+introducing changes into **core**, which affects **windows**, which is included in
+**main**)
+
+### Manual breakpoints
+
+If you want even more speed and variable/context inspection at the same time with the
+development itself, you can either traditionally debug the project by running the
+debugger or you can use manual breakpoints. They will stop the execution and spawn a
+Python shell for you to continue/inspect that code onwards. This shell can be exited
+(and execution resumes) with `Ctrl+D`/`Ctrl+Z`.
+
+You can place such "breakpoint" in either the library code itself or a Python
+unit-test:
+```python
+from RPA.core.helpers import interact
+interact(local={**globals(), **locals()})
+```
+
+## Release
```

Comment:
Because these instructions are for both internal and external contributors you may want to make a node here that the following sections only apply to Robocorp employees as external individuals do not have merge privileges.